pkg/build/strategies/onbuild/onbuild.go

```diff
@@ -0,0 +1,133 @@
+package onbuild
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/golang/glog"
+	"github.com/openshift/source-to-image/pkg/api"
+	"github.com/openshift/source-to-image/pkg/build"
+	"github.com/openshift/source-to-image/pkg/docker"
+	"github.com/openshift/source-to-image/pkg/git"
+	"github.com/openshift/source-to-image/pkg/tar"
+	"github.com/openshift/source-to-image/pkg/util"
+)
+
+// OnBuild strategy executes the simple Docker build in case the image does not
+// support STI scripts but has ONBUILD instructions recorded.
+type OnBuild struct {
+	docker  docker.Docker
+	git     git.Git
```

Comment:
I'm still thinking about interfacing VCS access as well, but that can be left until we'll actually support something more than just git, I guess.